### PR TITLE
golang sdk 0.1.23 rns added

### PR DESCRIPTION
**Pull request by jenniferopal**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \ Adding Golang SDK release notes 0.1.23 
* Jira/GitHub Issue numbers (if any): \ FFM-11482
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [X] Successful preview build.
- [X] Code owner review.
- [X] No merge conflicts.
- [X] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: release-notes/feature-flags.md [hunk 1]
```diff
@@ -15,12 +15,24 @@ Review the notes below for details about recent changes to Harness Feature Flags
...
-#### Last updated: May 9, 2024
+#### Last updated: May 16, 2024
 
 ## May 2024
 
 ### Golang SDK
 
+#### Version 0.1.23
+
+*New features and enhancements*:
```
Comment (richardblack-Harness) [inline release-notes/feature-flags.md:26]: Not a game changer, but should your headings be italic?

**Review by richardblack-Harness — APPROVED** [review]
One very minor detail regarding italic headings. Otherwise LGTM!

File: release-notes/feature-flags.md (hunk 1 above)
Comment (jenniferopal) [reply release-notes/feature-flags.md:26]: Oh shoot. They shouldn't. I think I've made this error a couple of times and didn't even notice. I'll correct that first. Thank you!

File: release-notes/feature-flags.md [hunk 2]
```diff
@@ -15,12 +15,24 @@ Review the notes below for details about recent changes to Harness Feature Flags
...
+
+**New features and enhancements**:
+ - Previously, from versions `v0.1.21` to `v0.1.22`, a critical bug was identified where evaluation metrics processing could cause a system panic. This occurred when at least one evaluation involved a target with `nil attributes` — either due to attributes being omitted or explicitly set to nil. This release addresses this issue by implementing a fix that prevents such panics. (FFM-11470, ZD-63281)
+
+**Enhanced Testing**: 
+ - To ensure the stability and reliability of this fix and similar functionalities, we have expanded the evaluation metrics suite of unit tests. These additional tests will help safeguard against regressions and improve code coverage.
+
+**Fixed issues**:
+ - Versions `v0.1.21` and `v0.1.22` have been retracted.
+ - Removes GitHub Actions step `Static Code Analysis / Build (push)` as it is constantly failing due to out of date credentials. These checks have moved to Harness CI but this step had yet to be removed until now.
```
Comment (douglas-j-bothwell) [inline release-notes/feature-flags.md:34]:
```suggestion
 - Removes GitHub Actions step `Static Code Analysis / Build (push)` as it is constantly failing due to out-of-date credentials. These checks have moved to Harness CI but this step had yet to be removed until now.
```

**Review by douglas-j-bothwell — APPROVED** [review]
One sug but otherwise LGTM

**Pull request merged**